### PR TITLE
Windows native pdb writers

### LINKED ISSUE #235: Consider removing "debugType": "portable" from project.json
Forcing the debug type to "portable" prevents VS' debugger from working correctly (e.g some variables can't be introspected).

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #235 

- Use the `Microsoft.DiaSymReader.Native` package.
- So, on dotnet publish related native assets are outputed to the runtime directory.
- During development, at runtime we do it ourselves from the package storage.
- A `csc.deps.json` file is created to allow the compiler to find the pdb writers.

**Logic**
- As dotnet, if `debugType` is not defined and on a windows platform, `full` is used.
- If `debugType` is not defined and not on a windows platform, `portable` is used.
- If `debugType` is defined it is used as it is defined without checking the platform.

**Rule**

- So, to be cross platforms it seems that the rule would be to never use explicitly `full`.
- Then we would have to choice between define `portable` or nothing.

I will do more tests.

### CODE REVIEW

File: src/Orchard.Environment.Extensions/Compilers/CSharpExtensionCompiler.cs [hunk 1]
```diff
@@ -44,6 +46,12 @@ private static RuntimeLibrary GetCscLibrary()
                     p => p.IndexOf("csc.exe", StringComparison.OrdinalIgnoreCase) >= 0))).FirstOrDefault();
         }
 
+        private static RuntimeLibrary GetNativePDBWriter()
+        {
+            return DependencyContext.Default?.RuntimeLibraries.Where(l =>
```
Comment (Jetski5822) [inline src/Orchard.Environment.Extensions/Compilers/CSharpExtensionCompiler.cs]: Instead of .Where().FirstOrDefault() use just .FirstOrDefault()

**Comment by jtkech** [comment]
So, seems to work as intended.

- I also removed the remaining `Orchard.Web` folder, let me know if not correct.

- For testing, remove `"debugType": "portable"` in a module project.json file, this to use (if on a windows platform) the `full` option (which needs native pdb writers). Then, see if dyn compilation doesn't fail.

- Note: on a non windows platform, the `portable` option is still used as dotnet do.

- Note: on a windows platform but using VS Code, right now the debugger can't read `full` debug infos.

Best.

**Comment by jtkech** [comment]
@sebastienros can we merge this PR? Works as before unless you remove `"debugType": "portable"` in the project.json of a module. In this case the `full` option is used and you can check that dyn compilation correctly uses native pdb writers (more details above). Also done some small refactorings and corrections.

**Comment by jtkech** [comment]
@sebastienros not urgent at all for me but just to know if you have seen my previous comments, and what's the rule to follow to be sure i'm not waiting for nothing.

- Do i need your agreement to merge a PR? If so i will not overcome this rule.
- If not needed, thanks, i will not abuse of it, but you know breaking changes are always possible.

Best.

**Comment by sebastienros** [comment]
You can merge when you think it's stable enough. Here I assume I will have merge conflicts but they are minor.

**Comment by jtkech** [comment]
Ok, thanks, first i'll rebase it on the last master branch and test it again.

**Pull request merged**